src/data/markdown/translated-guides/en/02 Using k6/14 Scenarios/00 About scenarios/02 VU allocation.md

```diff
@@ -0,0 +1,102 @@
+---
+title: VU allocation
+excerpt: How k6 allocates VUs in the open-model, arrival-rate executors
+---
+
+In arrival-rate executors, as long as k6 has VUs available, it starts iterations according to your target rate.
+The ability to set iteration rate comes with a bit more configuration complexity: you must pre-allocate a sufficient number of VUs.
+In other words, before the tests runs, you must both:
+- Configure load (as iteration per unit of time)
+- Ensure that you've scheduled enough VUs.
+
+Read on to learn about how k6 allocates VUs in the arrival-rate executors.
+
+## Pre-allocation in arrival-rate executors
+
+As [open-model](/using-k6/scenarios/about-scenarios/open-vs-closed/#open-model) scenarios, arrival-rate executors start iterations according to a configured rate.
+For example, you can configure arrival-rate executors to start 10 iterations each second, or minute, or hour.
+This behavior is opposed to the closed-model scenarios, in which VUs wait for one iteration to finish before starting another
+
+Each iteration need needs a VU to run it.
+Because k6 VUs are single threaded, like other JS runtimes, a VU can only run the event loop of a single iteration at a time.
+To ensure you have enough, you must pre-allocate a sufficient number.
+
+In your arrival-rate configuration, three properties determine the iteration rate:
+- `rate` determines how many iterations k6 starts.
+- `timeUnit` determines how frequently it starts the number of iterations.
+- `preAllocatedVUs` sets the number of VUs to use to reach the target iterations per second.
+In practice, determining the right number of iterations might take some trial and error,
+as the necessary VUs entirely depends on how quickly the SUT can process iterations.
+
+```javascript
+export const options = {
+  scenarios: {
+    constant_load: {
+      executor: "constant-arrival-rate",
+      preAllocatedVUs: 4,
+      rate: 8,
+      timeUnit: "1m",
+    },
+  },
+};
+```
+
+
+<Blockquote mod="attention" title="">
+
+In cloud tests, **both `preAllocatedVUs` and `maxVUs` count against your subscription.**
```

Comment:
Another problem - you mention `maxVUs` here, but we haven't mentioned maxVUs anywhere before in this document... This was one of the reasons for my longer explanation that you discarded, it had a paragraph with a cohesive explanation for both `preAllocatedVUs` and `maxVUs`

Reply:
I'm still kind of of the opinion that `maxVUs` shouldn't be mentioned anywhere :-). What I'll do is just make the first admonition only about `preAllocatedVUs`, and then a second admonition in the `maxVUs` section. It's not very elegant, but it doesn't cram so much information together.

Reply:
Again, if I could rewrite history, maxVUs probably would not exist :sweat_smile:  But now that it exists, we should try to make it as clear as possible how it function and why it might not be a good idea to use. 

An admonition only for `preAllocatedVUs` doesn't make sense, this is how cloud subscriptions normally work, i.e. no admonition needed just for it.

And if we want to tuck `maxVUs` only at the end of the document, somewhat out of sight (which I don't necessarily mind), then we should only have an admonition there.

Reply:
Here's how I've handled it: two admonitions. No points for subtlety, but I don't think anyone can say we're being sneaky:

![image](https://user-images.githubusercontent.com/47385188/212304282-b337ec55-3b54-409d-b466-ffa325471d49.png)


![image](https://user-images.githubusercontent.com/47385188/212304169-5155db43-48a6-4a72-8908-32d544c96357.png)

Reply:
Nobody will read the text you have highlighted in the middle of that paragraph :sweat_smile: People will just see the two admonitions and get the impressions that in the cloud we will charge them for `preAllocatedVUs + maxVUs`

Reply:
Point taken. With https://github.com/grafana/k6-docs/pull/974/commits/19784d4173ca6376585990b447f1b39cbcc58706, it looks like this:

![image](https://user-images.githubusercontent.com/47385188/212308372-f85b547e-549d-4d40-9845-3b748d8776ac.png)